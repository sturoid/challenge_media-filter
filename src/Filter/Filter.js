import React, { useState, useEffect, useRef } from 'react';
import Fuse from 'fuse.js';
import { typeMedia, typeArrayOf, typeString } from '../lib/prop-types';
import { Wrapper, Grid, Button, SelectMulti } from '../shared';
import { arrayToUpperCase } from './Filter.utils';
import './Filter.scss';

function filterData(data, selectedType, selectedYears, selectedGenres, searchString) {
  if (!selectedType && !selectedYears.length && !selectedGenres.length && !searchString)
    return data;
  let newData = data;

  if (selectedType) {
    newData = newData.filter(({ type }) => {
      return type === selectedType;
    });
  }

  if (selectedYears.length) {
    newData = newData.filter(({ year }) => {
      return selectedYears.includes(year);
    });
  }

  if (selectedGenres.length) {
    newData = newData.filter(({ genre }) => {
      return selectedGenres.some(selected => genre.includes(selected));
    });
  }

  if (searchString) {
    const fuse = new Fuse(newData, { keys: ['title'] });
    newData = fuse.search(searchString);
  }

  return newData;
}

const Filter = ({ data, years, genres }) => {
  const [filteredData, setFilterData] = useState(data);
  const [selectedType, setSelectedType] = useState(null);
  const [selectedYears, setSelectedYears] = useState([]);
  const [selectedGenres, setSelectedGenres] = useState([]);
  const [searchString, setSearchString] = useState(null);
  const searchRef = useRef(null);

  useEffect(() => {
    const newData = filterData(
      data,
      selectedType,
      selectedYears,
      selectedGenres,
      searchString
    );
    setFilterData(newData);
  }, [selectedType, selectedYears, selectedGenres, searchString]);

  function filterType({ target: { value } }) {
    setSelectedType(value.toLowerCase().substr(0, value.length - 1));
  }

  function filterYears(yearsFromSelect) {
    setSelectedYears(yearsFromSelect);
  }

  function filterGenres(genresFromSelect) {
    setSelectedGenres(genresFromSelect.map(g => g.toLowerCase()));
  }

  function filterTitle({ target: { value } }) {
    setSearchString(value);
  }

  function clearFilters() {
    setFilterData(data);
    setSelectedType(null);
    setSelectedYears([]);
    setSelectedGenres([]);
    setSearchString(null);
    setSelectedType(null);
    searchRef.current.value = '';
  }

  return (
    <Wrapper>
      <div className="filter-head">
        <Grid.Row>
          <Grid.Left>
            <SelectMulti
              options={years}
              callback={filterYears}
              reset={!selectedYears.length}
            />
            <SelectMulti
              options={genres}
              callback={filterGenres}
              reset={!selectedGenres.length}
            />
          </Grid.Left>

          <Grid.Right>
            <input type="search" name="search" ref={searchRef} onChange={filterTitle} />
          </Grid.Right>
        </Grid.Row>

        <Grid.Row>
          <Grid.Left>
            <div className="left">
              <label htmlFor="type">
                <input
                  type="radio"
                  name="type"
                  checked={selectedType === 'movie'}
                  value="Movies"
                  onChange={filterType}
                />
                Movies
              </label>
              <label htmlFor="type">
                <input
                  type="radio"
                  name="type"
                  checked={selectedType === 'book'}
                  value="Books"
                  onChange={filterType}
                />
                Books
              </label>
            </div>
          </Grid.Left>
          <Grid.Right>
            <div className="right">
              <Button asLink onClick={clearFilters}>
                Clear Filters
              </Button>
            </div>
          </Grid.Right>
        </Grid.Row>
      </div>

      <div className="filter-content">
        {filteredData.map(d => {
          return (
            <div key={d.title} className="filter-item">
              <img className="poster" src={d.poster} alt="poster" />
              <div className="title">{`${d.title} (${d.year})`}</div>
              <div className="genre-list">{arrayToUpperCase(d.genre).join(', ')}</div>
            </div>
          );
        })}
      </div>
    </Wrapper>
  );
};

Filter.propTypes = {
  data: typeMedia.isRequired,
  years: typeArrayOf(typeString).isRequired,
  genres: typeArrayOf(typeString).isRequired
};

export default Filter;
