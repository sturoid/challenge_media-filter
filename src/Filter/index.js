export { default } from './Filter.data';
