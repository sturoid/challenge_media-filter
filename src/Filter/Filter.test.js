import React from 'react';
import renderer from 'react-test-renderer';
import Filter from './Filter';

describe('Filter', () => {
  it('renders', () => {
    const component = renderer.create(<Filter />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
