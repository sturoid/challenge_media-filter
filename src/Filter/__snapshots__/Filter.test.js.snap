// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders 1`] = `
<div
  className="wrapper"
>
  <div
    className="filter"
  >
    <div
      className="filter-head"
    >
      <div
        className="row"
      >
        <div
          className="row-left"
        >
          <div
            className="select-multi"
          >
            <div
              className="placeholder"
              onClick={[Function]}
              role="button"
              tabIndex="-1"
            >
              <div
                className="text"
              >
                Genre
              </div>
              <div
                className="icon"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="6 9 12 15 18 9"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            className="select-multi"
          >
            <div
              className="placeholder"
              onClick={[Function]}
              role="button"
              tabIndex="-1"
            >
              <div
                className="text"
              >
                Year
              </div>
              <div
                className="icon"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="6 9 12 15 18 9"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          className="row-right"
        >
          <div
            className="search"
          >
            <input
              name="search"
              onChange={[Function]}
            />
            <div
              className="icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="row-left"
        >
          <div
            className="radio"
          >
            <label
              htmlFor="movie"
            >
              <input
                checked={false}
                id="movie"
                name="type"
                onChange={[Function]}
                type="radio"
                value="movie"
              />
              Movies
            </label>
          </div>
          <div
            className="radio"
          >
            <label
              htmlFor="book"
            >
              <input
                checked={false}
                id="book"
                name="type"
                onChange={[Function]}
                type="radio"
                value="book"
              />
              Books
            </label>
          </div>
        </div>
        <div
          className="row-right"
        >
          <button
            className="btn primary text-uppercase as-link"
            onClick={[Function]}
            type="button"
          >
            Clear Filters
          </button>
        </div>
      </div>
    </div>
    <div
      className="filter-content"
    >
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BMjA0ODEzMTc1Nl5BMl5BanBnXkFtZTcwODM2MjAxNA@@._V1_SY1000_CR0,0,664,1000_AL_.jpg"
        />
        <div
          className="title"
        >
          Raiders of the Lost Ark (1981)
        </div>
        <div
          className="genre-list"
        >
          Action, Adventure
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BMTc0NDQzNTA2Ml5BMl5BanBnXkFtZTcwNzI2OTQzMw@@._V1_.jpg"
        />
        <div
          className="title"
        >
          The Other Guys (2010)
        </div>
        <div
          className="genre-list"
        >
          Action, Comedy, Crime
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BMjM2MDgxMDg0Nl5BMl5BanBnXkFtZTgwNTM2OTM5NDE@._V1_.jpg"
        />
        <div
          className="title"
        >
          Jurassic Park (1993)
        </div>
        <div
          className="genre-list"
        >
          Adventure, Sci-fi, Thriller
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BOTIyMDY2NGQtOGJjNi00OTk4LWFhMDgtYmE3M2NiYzM0YTVmXkEyXkFqcGdeQXVyNTU1NTcwOTk@._V1_SY1000_CR0,0,654,1000_AL_.jpg"
        />
        <div
          className="title"
        >
          Star Wars: A New Hope (1977)
        </div>
        <div
          className="genre-list"
        >
          Action, Adventure, Fantasy
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BMTI2ODMzODA0Ml5BMl5BanBnXkFtZTYwNTM3NzY5._V1._CR17,27,308,447_.jpg"
        />
        <div
          className="title"
        >
          Predator (1987)
        </div>
        <div
          className="genre-list"
        >
          Action, Adventure, Sci-fi
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BMTcyMzQ0MjQ5MF5BMl5BanBnXkFtZTYwOTczNjA5._V1_.jpg"
        />
        <div
          className="title"
        >
          Blazing Saddles (1974)
        </div>
        <div
          className="genre-list"
        >
          Comedy, Western
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BN2YyNDE5NzItMjAwNC00MGQxLTllNjktZGIzMWFkZjA3OWQ0XkEyXkFqcGdeQXVyNDk3NzU2MTQ@._V1_SY1000_CR0,0,699,1000_AL_.jpg"
        />
        <div
          className="title"
        >
          Chinatown (1974)
        </div>
        <div
          className="genre-list"
        >
          Drama, Mystery, Thriller
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BNzMyMjQ2OTEwOF5BMl5BanBnXkFtZTgwNzc2OTkwMDE@._V1_.jpg"
        />
        <div
          className="title"
        >
          Beverly Hills Cop (1984)
        </div>
        <div
          className="genre-list"
        >
          Action, Comedy, Crime
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BMTkzODMwNDkzOF5BMl5BanBnXkFtZTgwNDA4NzA1ODE@._V1_SY1000_CR0,0,687,1000_AL_.jpg"
        />
        <div
          className="title"
        >
          The Invitation (2015)
        </div>
        <div
          className="genre-list"
        >
          Horror, Thriller
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BYzczOGRlMzQtNDAzMS00MjdlLTk5Y2QtNTM3MDE3NjRkYzQwXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_.jpg"
        />
        <div
          className="title"
        >
          The Exorcist (1973)
        </div>
        <div
          className="genre-list"
        >
          Horror
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BMTQxNTIzMTExN15BMl5BanBnXkFtZTcwODI4MDgzNA@@._V1_SY1000_CR0,0,666,1000_AL_.jpg"
        />
        <div
          className="title"
        >
          From Russia With Love (1963)
        </div>
        <div
          className="genre-list"
        >
          Action, Adventure, Thriller
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BMjEyMzgwNTUzMl5BMl5BanBnXkFtZTcwNTMxMzM3Ng@@._V1_SY1000_CR0,0,772,1000_AL_.jpg"
        />
        <div
          className="title"
        >
          Lion King (1994)
        </div>
        <div
          className="genre-list"
        >
          Animation, Adventure, Drama
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BMTI4MDAwMDY3N15BMl5BanBnXkFtZTcwODIwMzMzMQ@@._V1._CR46,1,342,473_.jpg"
        />
        <div
          className="title"
        >
          Terminator 2: Judgment Day (1991)
        </div>
        <div
          className="genre-list"
        >
          Action, Sci-fi
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BMTk1MDUzMTQ3OV5BMl5BanBnXkFtZTcwMDAwNTk0NA@@._V1_SY1000_CR0,0,666,1000_AL_.jpg"
        />
        <div
          className="title"
        >
          Robocop (1987)
        </div>
        <div
          className="genre-list"
        >
          Action, Crime, Sci-fi
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://ia.media-imdb.com/images/M/MV5BMTY1MTg1NDAxOV5BMl5BanBnXkFtZTcwMjg1MDI5Nw@@._V1_SY1000_CR0,0,675,1000_AL_.jpg"
        />
        <div
          className="title"
        >
          Finding Nemo (2003)
        </div>
        <div
          className="genre-list"
        >
          Animation, Adventure, Comedy
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1327868566l/2429135.jpg"
        />
        <div
          className="title"
        >
          The Girl with the Dragon Tattoo (2005)
        </div>
        <div
          className="genre-list"
        >
          Crime, Mystery, Thriller
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://i.kinja-img.com/gawker-media/image/upload/s---ohdLyWR--/17m13vh1sfoq3jpg.jpg"
        />
        <div
          className="title"
        >
          Roadside Picnic (1971)
        </div>
        <div
          className="genre-list"
        >
          Sci-fi, Fantasy
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1392015757l/1971304.jpg"
        />
        <div
          className="title"
        >
          City of Thieves (2008)
        </div>
        <div
          className="genre-list"
        >
          Adventure, History, War
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1403345411l/18144124.jpg"
        />
        <div
          className="title"
        >
          I Am Pilgrim (2013)
        </div>
        <div
          className="genre-list"
        >
          Mystery, Thriller, War
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1303252999l/968.jpg"
        />
        <div
          className="title"
        >
          The Da Vinci Code (2003)
        </div>
        <div
          className="genre-list"
        >
          Adventure, Crime, Mystery
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1433086293l/21.jpg"
        />
        <div
          className="title"
        >
          A Short History of Nearly Everything (2003)
        </div>
        <div
          className="genre-list"
        >
          Comedy, History, Science
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1433739086l/33313.jpg"
        />
        <div
          className="title"
        >
          Kitchen Confidential (2000)
        </div>
        <div
          className="genre-list"
        >
          Biography, Food & drink, Memoir
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1348621563l/347610.jpg"
        />
        <div
          className="title"
        >
          King Leopold's Ghost (1998)
        </div>
        <div
          className="genre-list"
        >
          Biography, History, Politics
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1414608893l/2715.jpg"
        />
        <div
          className="title"
        >
          Salt Book (2002)
        </div>
        <div
          className="genre-list"
        >
          Food & drink, History, Science
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1409498796l/22836957.jpg"
        />
        <div
          className="title"
        >
          The Revenant (2002)
        </div>
        <div
          className="genre-list"
        >
          History, Thriller, Western
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1436732693l/13496.jpg"
        />
        <div
          className="title"
        >
          A Game of Thrones (1996)
        </div>
        <div
          className="genre-list"
        >
          Adventure, Dragons, Fantasy
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1327942880l/2493.jpg"
        />
        <div
          className="title"
        >
          The Time Machine (1895)
        </div>
        <div
          className="genre-list"
        >
          Adventure, Classics, Sci-fi
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1345271681l/33507.jpg"
        />
        <div
          className="title"
        >
          20,000 Leagues Under the Sea (1870)
        </div>
        <div
          className="genre-list"
        >
          Adventure, Classics, Sci-fi
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1164045516l/3590.jpg"
        />
        <div
          className="title"
        >
          The Adventures of Sherlock Holmes (1892)
        </div>
        <div
          className="genre-list"
        >
          Classics, Detective, Mystery
        </div>
      </div>
      <div
        className="filter-item"
      >
        <img
          alt="poster"
          className="poster"
          src="https://d.gr-assets.com/books/1443231179l/12497.jpg"
        />
        <div
          className="title"
        >
          No Country For Old Men (2005)
        </div>
        <div
          className="genre-list"
        >
          Mystery, Thriller, Western
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Filter.utils #filterData filters by genre 1`] = `
Array [
  Object {
    "genre": Array [
      "action",
      "adventure",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMjA0ODEzMTc1Nl5BMl5BanBnXkFtZTcwODM2MjAxNA@@._V1_SY1000_CR0,0,664,1000_AL_.jpg",
    "title": "Raiders of the Lost Ark",
    "type": "movie",
    "year": "1981",
  },
  Object {
    "genre": Array [
      "action",
      "comedy",
      "crime",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTc0NDQzNTA2Ml5BMl5BanBnXkFtZTcwNzI2OTQzMw@@._V1_.jpg",
    "title": "The Other Guys",
    "type": "movie",
    "year": "2010",
  },
  Object {
    "genre": Array [
      "action",
      "adventure",
      "fantasy",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BOTIyMDY2NGQtOGJjNi00OTk4LWFhMDgtYmE3M2NiYzM0YTVmXkEyXkFqcGdeQXVyNTU1NTcwOTk@._V1_SY1000_CR0,0,654,1000_AL_.jpg",
    "title": "Star Wars: A New Hope",
    "type": "movie",
    "year": "1977",
  },
  Object {
    "genre": Array [
      "action",
      "adventure",
      "sci-fi",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTI2ODMzODA0Ml5BMl5BanBnXkFtZTYwNTM3NzY5._V1._CR17,27,308,447_.jpg",
    "title": "Predator",
    "type": "movie",
    "year": "1987",
  },
  Object {
    "genre": Array [
      "action",
      "comedy",
      "crime",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BNzMyMjQ2OTEwOF5BMl5BanBnXkFtZTgwNzc2OTkwMDE@._V1_.jpg",
    "title": "Beverly Hills Cop",
    "type": "movie",
    "year": "1984",
  },
  Object {
    "genre": Array [
      "action",
      "adventure",
      "thriller",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTQxNTIzMTExN15BMl5BanBnXkFtZTcwODI4MDgzNA@@._V1_SY1000_CR0,0,666,1000_AL_.jpg",
    "title": "From Russia With Love",
    "type": "movie",
    "year": "1963",
  },
  Object {
    "genre": Array [
      "action",
      "sci-fi",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTI4MDAwMDY3N15BMl5BanBnXkFtZTcwODIwMzMzMQ@@._V1._CR46,1,342,473_.jpg",
    "title": "Terminator 2: Judgment Day",
    "type": "movie",
    "year": "1991",
  },
  Object {
    "genre": Array [
      "action",
      "crime",
      "sci-fi",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTk1MDUzMTQ3OV5BMl5BanBnXkFtZTcwMDAwNTk0NA@@._V1_SY1000_CR0,0,666,1000_AL_.jpg",
    "title": "Robocop",
    "type": "movie",
    "year": "1987",
  },
]
`;

exports[`Filter.utils #filterData filters by genre 2`] = `
Array [
  Object {
    "genre": Array [
      "action",
      "adventure",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMjA0ODEzMTc1Nl5BMl5BanBnXkFtZTcwODM2MjAxNA@@._V1_SY1000_CR0,0,664,1000_AL_.jpg",
    "title": "Raiders of the Lost Ark",
    "type": "movie",
    "year": "1981",
  },
  Object {
    "genre": Array [
      "action",
      "comedy",
      "crime",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTc0NDQzNTA2Ml5BMl5BanBnXkFtZTcwNzI2OTQzMw@@._V1_.jpg",
    "title": "The Other Guys",
    "type": "movie",
    "year": "2010",
  },
  Object {
    "genre": Array [
      "action",
      "adventure",
      "fantasy",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BOTIyMDY2NGQtOGJjNi00OTk4LWFhMDgtYmE3M2NiYzM0YTVmXkEyXkFqcGdeQXVyNTU1NTcwOTk@._V1_SY1000_CR0,0,654,1000_AL_.jpg",
    "title": "Star Wars: A New Hope",
    "type": "movie",
    "year": "1977",
  },
  Object {
    "genre": Array [
      "action",
      "adventure",
      "sci-fi",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTI2ODMzODA0Ml5BMl5BanBnXkFtZTYwNTM3NzY5._V1._CR17,27,308,447_.jpg",
    "title": "Predator",
    "type": "movie",
    "year": "1987",
  },
  Object {
    "genre": Array [
      "comedy",
      "western",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTcyMzQ0MjQ5MF5BMl5BanBnXkFtZTYwOTczNjA5._V1_.jpg",
    "title": "Blazing Saddles",
    "type": "movie",
    "year": "1974",
  },
  Object {
    "genre": Array [
      "action",
      "comedy",
      "crime",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BNzMyMjQ2OTEwOF5BMl5BanBnXkFtZTgwNzc2OTkwMDE@._V1_.jpg",
    "title": "Beverly Hills Cop",
    "type": "movie",
    "year": "1984",
  },
  Object {
    "genre": Array [
      "action",
      "adventure",
      "thriller",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTQxNTIzMTExN15BMl5BanBnXkFtZTcwODI4MDgzNA@@._V1_SY1000_CR0,0,666,1000_AL_.jpg",
    "title": "From Russia With Love",
    "type": "movie",
    "year": "1963",
  },
  Object {
    "genre": Array [
      "action",
      "sci-fi",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTI4MDAwMDY3N15BMl5BanBnXkFtZTcwODIwMzMzMQ@@._V1._CR46,1,342,473_.jpg",
    "title": "Terminator 2: Judgment Day",
    "type": "movie",
    "year": "1991",
  },
  Object {
    "genre": Array [
      "action",
      "crime",
      "sci-fi",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTk1MDUzMTQ3OV5BMl5BanBnXkFtZTcwMDAwNTk0NA@@._V1_SY1000_CR0,0,666,1000_AL_.jpg",
    "title": "Robocop",
    "type": "movie",
    "year": "1987",
  },
  Object {
    "genre": Array [
      "animation",
      "adventure",
      "comedy",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTY1MTg1NDAxOV5BMl5BanBnXkFtZTcwMjg1MDI5Nw@@._V1_SY1000_CR0,0,675,1000_AL_.jpg",
    "title": "Finding Nemo",
    "type": "movie",
    "year": "2003",
  },
  Object {
    "genre": Array [
      "comedy",
      "history",
      "science",
    ],
    "poster": "https://d.gr-assets.com/books/1433086293l/21.jpg",
    "title": "A Short History of Nearly Everything",
    "type": "book",
    "year": "2003",
  },
]
`;

exports[`Filter.utils #filterData filters by search string 1`] = `
Array [
  Object {
    "genre": Array [
      "action",
      "adventure",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMjA0ODEzMTc1Nl5BMl5BanBnXkFtZTcwODM2MjAxNA@@._V1_SY1000_CR0,0,664,1000_AL_.jpg",
    "title": "Raiders of the Lost Ark",
    "type": "movie",
    "year": "1981",
  },
  Object {
    "genre": Array [
      "sci-fi",
      "fantasy",
    ],
    "poster": "https://i.kinja-img.com/gawker-media/image/upload/s---ohdLyWR--/17m13vh1sfoq3jpg.jpg",
    "title": "Roadside Picnic",
    "type": "book",
    "year": "1971",
  },
  Object {
    "genre": Array [
      "adventure",
      "sci-fi",
      "thriller",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMjM2MDgxMDg0Nl5BMl5BanBnXkFtZTgwNTM2OTM5NDE@._V1_.jpg",
    "title": "Jurassic Park",
    "type": "movie",
    "year": "1993",
  },
]
`;

exports[`Filter.utils #filterData filters by search string 2`] = `
Array [
  Object {
    "genre": Array [
      "action",
      "sci-fi",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTI4MDAwMDY3N15BMl5BanBnXkFtZTcwODIwMzMzMQ@@._V1._CR46,1,342,473_.jpg",
    "title": "Terminator 2: Judgment Day",
    "type": "movie",
    "year": "1991",
  },
  Object {
    "genre": Array [
      "action",
      "adventure",
      "sci-fi",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTI2ODMzODA0Ml5BMl5BanBnXkFtZTYwNTM3NzY5._V1._CR17,27,308,447_.jpg",
    "title": "Predator",
    "type": "movie",
    "year": "1987",
  },
  Object {
    "genre": Array [
      "drama",
      "mystery",
      "thriller",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BN2YyNDE5NzItMjAwNC00MGQxLTllNjktZGIzMWFkZjA3OWQ0XkEyXkFqcGdeQXVyNDk3NzU2MTQ@._V1_SY1000_CR0,0,699,1000_AL_.jpg",
    "title": "Chinatown",
    "type": "movie",
    "year": "1974",
  },
  Object {
    "genre": Array [
      "comedy",
      "history",
      "science",
    ],
    "poster": "https://d.gr-assets.com/books/1433086293l/21.jpg",
    "title": "A Short History of Nearly Everything",
    "type": "book",
    "year": "2003",
  },
]
`;

exports[`Filter.utils #filterData filters by search string 3`] = `
Array [
  Object {
    "genre": Array [
      "action",
      "crime",
      "sci-fi",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTk1MDUzMTQ3OV5BMl5BanBnXkFtZTcwMDAwNTk0NA@@._V1_SY1000_CR0,0,666,1000_AL_.jpg",
    "title": "Robocop",
    "type": "movie",
    "year": "1987",
  },
  Object {
    "genre": Array [
      "mystery",
      "thriller",
      "western",
    ],
    "poster": "https://d.gr-assets.com/books/1443231179l/12497.jpg",
    "title": "No Country For Old Men",
    "type": "book",
    "year": "2005",
  },
  Object {
    "genre": Array [
      "action",
      "comedy",
      "crime",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BNzMyMjQ2OTEwOF5BMl5BanBnXkFtZTgwNzc2OTkwMDE@._V1_.jpg",
    "title": "Beverly Hills Cop",
    "type": "movie",
    "year": "1984",
  },
]
`;

exports[`Filter.utils #filterData filters by year 1`] = `
Array [
  Object {
    "genre": Array [
      "action",
      "adventure",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMjA0ODEzMTc1Nl5BMl5BanBnXkFtZTcwODM2MjAxNA@@._V1_SY1000_CR0,0,664,1000_AL_.jpg",
    "title": "Raiders of the Lost Ark",
    "type": "movie",
    "year": "1981",
  },
]
`;

exports[`Filter.utils #filterData filters by year 2`] = `
Array [
  Object {
    "genre": Array [
      "action",
      "adventure",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMjA0ODEzMTc1Nl5BMl5BanBnXkFtZTcwODM2MjAxNA@@._V1_SY1000_CR0,0,664,1000_AL_.jpg",
    "title": "Raiders of the Lost Ark",
    "type": "movie",
    "year": "1981",
  },
  Object {
    "genre": Array [
      "action",
      "comedy",
      "crime",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTc0NDQzNTA2Ml5BMl5BanBnXkFtZTcwNzI2OTQzMw@@._V1_.jpg",
    "title": "The Other Guys",
    "type": "movie",
    "year": "2010",
  },
]
`;

exports[`Filter.utils #filterData filters multiple 1`] = `
Array [
  Object {
    "genre": Array [
      "action",
      "sci-fi",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTI4MDAwMDY3N15BMl5BanBnXkFtZTcwODIwMzMzMQ@@._V1._CR46,1,342,473_.jpg",
    "title": "Terminator 2: Judgment Day",
    "type": "movie",
    "year": "1991",
  },
]
`;

exports[`Filter.utils #filterData filters multiple 2`] = `
Array [
  Object {
    "genre": Array [
      "horror",
      "thriller",
    ],
    "poster": "https://ia.media-imdb.com/images/M/MV5BMTkzODMwNDkzOF5BMl5BanBnXkFtZTgwNDA4NzA1ODE@._V1_SY1000_CR0,0,687,1000_AL_.jpg",
    "title": "The Invitation",
    "type": "movie",
    "year": "2015",
  },
]
`;

exports[`Filter.utils #filterData filters multiple 3`] = `Array []`;
